v2/controllers/marketplace/meterbase_controller.go

```diff
@@ -265,6 +297,68 @@ func (r *MeterBaseReconciler) Reconcile(request reconcile.Request) (reconcile.Re
 		instance.Status.Conditions = status.Conditions{}
 	}
 
+	// userWorkloadMonitoringEnabled is considered enabled if both the Spec and cluster configuration are satisfied
+	userWorkloadMonitoringEnabledSpec := instance.Spec.UserWorkloadMonitoringEnabled
+	userWorkloadMonitoringEnabledOnCluster, result := isUserWorkloadMonitoringEnabledOnCluster(cc, r.cfg.Infrastructure, reqLogger)
+	if !result.Is(Continue) {
+		return result.Return()
+	}
+	userWorkloadMonitoringEnabled := userWorkloadMonitoringEnabledOnCluster && userWorkloadMonitoringEnabledSpec
+
+	if instance.Status.Conditions.IsUnknownFor(marketplacev1alpha1.ConditionUserWorkloadMonitoringEnabled) {
+		// Set initial UWM status
+		if result, err := updateUserWorkloadMonitoringEnabledStatus(cc,
```

Comment:
Don't need an if here. We should requeue regardless. Only if condition should be if err after result is returned.

Reply:
This is intentional. Do not want to updateUserWorkloadMonitoringEnabledStatus if the state is Transitioning, until the transition timer has expired.